src/test/rpc/ValidatorRPC_test.cpp

```diff
@@ -0,0 +1,377 @@
+
+        auto toStr = [](PublicKey const& publicKey) {
+            return toBase58(TokenType::TOKEN_NODE_PUBLIC, publicKey);
+        };
+
+        // Publisher manifest/signing keys
+        auto const publisherSecret = randomSecretKey();
+        auto const publisherPublic =
+            derivePublicKey(KeyType::ed25519, publisherSecret);
+        auto const publisherSigningKeys = randomKeyPair(KeyType::secp256k1);
+        auto const manifest = makeManifestString(
+            publisherPublic,
+            publisherSecret,
+            publisherSigningKeys.first,
+            publisherSigningKeys.second,
+            1);
+
+        // Validator keys that will be in the published list
+        std::vector<PublicKey> keys = {randomNode(), randomNode()};
+        std::set<std::string> expectedKeys;
+        for (auto const& key : keys)
+            expectedKeys.insert(toStr(key));
+
+
+        //----------------------------------------------------------------------
+        // Publisher list site unavailable
+        {
+            // Publisher site information
+            std::string siteURI = "http://127.0.0.1:1234/validators";
+
+            Env env{
+                *this,
+                envconfig([&](std::unique_ptr<Config> cfg) {
+                    cfg->section(SECTION_VALIDATOR_LIST_SITES).append(siteURI);
+                    cfg->section(SECTION_VALIDATOR_LIST_KEYS)
+                        .append(strHex(publisherPublic));
+                    return cfg;
+                }),
+            };
+            {
+                auto const jrr = env.rpc("server_info")[jss::result];
+                BEAST_EXPECT(
+                    jrr[jss::info][jss::validator_list_expires] == "unknown");
+            }
+            {
+                auto const jrr = env.rpc("server_state")[jss::result];
+                BEAST_EXPECT(
+                    jrr[jss::state][jss::validator_list_expires].asInt() == 0);
+            }
+            {
+                auto const jrr = env.rpc("validators")[jss::result];
+                BEAST_EXPECT(jrr[jss::validation_quorum].asUInt() ==
+                    std::numeric_limits<std::uint32_t>::max());
+                BEAST_EXPECT(jrr[jss::local_static_keys].size() == 0);
+                BEAST_EXPECT(jrr[jss::trusted_validator_keys].size() == 0);
+                BEAST_EXPECT(jrr[jss::validator_list_expires] == "unknown");
+
+                if (BEAST_EXPECT(jrr[jss::publisher_lists].size() == 1))
+                {
+                    auto jp = jrr[jss::publisher_lists][0u];
+                    BEAST_EXPECT(jp[jss::available] == false);
+                    BEAST_EXPECT(jp[jss::list].size() == 0);
+                    BEAST_EXPECT(jp[jss::seq].asUInt() == 0);
+                    BEAST_EXPECT(
+                        jp[jss::pubkey_publisher] == strHex(publisherPublic));
+                    BEAST_EXPECT(jp[jss::expiration] ==
+                        to_string(TimeKeeper::time_point{}));
```

Comment:
I'm wondering if `seq` and `expiration` should either be `"unknown"` or excluded in this situation

Reply:
Is there a way to distinguish a list that was previously loaded and is now unavailable versus one that has not been successfully loaded yet?

Reply:
The `expiration` would be `TimeKeeper::time_point{}` in the latter case. (We would never accept a list with that expiration without a time machine.)